pkg/filestore/writer_test.go

```diff
@@ -311,10 +385,30 @@ func Test_WriteKeypair(t *testing.T) {
 
 			testBundle := test.testBundle
 			err = w.WriteKeypair(test.meta, testBundle.pk, testBundle.certPEM, testBundle.caPEM)
-			assert.Equal(t, test.expErr, err != nil, "%v", err)
+
+			// make sure to return here if error was expected and received
+			// validation will prevent files from being written
+			if test.expErr {
```

Comment:
```suggestion
			assert.Equal(t, test.expErr, err != nil, "%v", err)
```

Reply:
Same here, we must stop the test here as further assertions are not possible if an error has been returned